integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/client/TopicSender.java

```diff
@@ -0,0 +1,104 @@
+import org.wso2.ei.mb.test.utils.JMSAcknowledgeMode;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Properties;
+import javax.jms.JMSException;
+import javax.jms.TextMessage;
+import javax.jms.Topic;
+import javax.jms.TopicConnection;
+import javax.jms.TopicConnectionFactory;
+import javax.jms.TopicPublisher;
+import javax.jms.TopicSession;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+
+/**
+ * Message sender to the topics.
+ */
+public class TopicSender {
+    private static Logger log = Logger.getLogger(TopicSender.class);
+
+    private TopicConnection topicConnection;
+    private TopicSession topicSession;
+    private TopicPublisher publisher;
+
+    /**
+     * Initialize the topic publisher.
+     * @param topicName Topic name to publish
+     * @param acknowledgeMode JMSAcknowledgement mode
+     * @param configurationReader Server configurations
+     * @throws IOException
+     * @throws NamingException
+     * @throws JMSException
+     */
+    public TopicSender(String topicName, JMSAcknowledgeMode acknowledgeMode,
+         ConfigurationReader configurationReader) throws IOException, NamingException, JMSException {
+
+        // map of config key and config value
+        Map<String, String> clientConfigPropertiesMap = configurationReader.getClientConfigProperties();
+        Properties properties = new Properties();
+        properties.put(Context.INITIAL_CONTEXT_FACTORY, ClientConstants.FILE_INITIAL_CONNECTION_FACTORY);
+        properties.put(ClientConstants.CONNECTION_FACTORY_NAME_PREFIX + ClientConstants.ANDES_CONNECTION_FACTORY_NAME,
+                ClientUtils.getTCPConnectionURL(clientConfigPropertiesMap));
+        properties.put(ClientConstants.TOPIC_NAME_PREFIX + topicName, topicName);
+        InitialContext ctx = new InitialContext(properties);
+
+        TopicConnectionFactory connectionFactory = (TopicConnectionFactory) ctx
+                .lookup(ClientConstants.ANDES_CONNECTION_FACTORY_NAME);
+        topicConnection = connectionFactory.createTopicConnection();
+        topicConnection.start();
+        topicSession = topicConnection.createTopicSession(false, acknowledgeMode.getType());
+        Topic topic = (Topic) ctx.lookup(topicName);
+        publisher = topicSession.createPublisher(topic);
+
+    }
+
+    /**
+     * Send a text message to the topic.
+     * @param messageCount Number of messages to send
+     * @param textPayload   Message content
+     * @throws JMSException
+     */
+    public void sendMessages(int messageCount, String textPayload) throws JMSException {
+        TextMessage message = topicSession.createTextMessage(textPayload);
+
+        for (int i = 0; i < messageCount; i++) {
+            publisher.publish(message);
+        }
+    }
+
+    /**
+     * Close the publisher connections.
+     * @throws JMSException
+     */
+    public void closeSender() throws JMSException {
```

Comment:
Inside closeReceiver, we have caught an NPE, don't we need it here?

Reply:
Added that NPE since it causes some NPEs inside close operation. Did not observe such a issue for this.